demo/c-api/c-api-demo.cc

```diff
@@ -0,0 +1,169 @@
+/*!
+ * Copyright 2019 XGBoost contributors
+ *
+ * \file c-api-demo.c
+ * \brief A simple example of using xgboost C API.
+ */
+
+#include <iostream>
+#include <fstream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <xgboost/c_api.h>
+
+#include <enclave/crypto.h>
+#include <openenclave/host.h>
+
+static char test_key[CIPHER_KEY_SIZE] = {
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0
+};
+
+#define safe_xgboost(call) {                                            \
+int err = (call);                                                       \
+if (err != 0) {                                                         \
+  fprintf(stderr, "%s:%d: error in %s: %s\n", __FILE__, __LINE__, #call, XGBGetLastError()); \
+  exit(1);                                                              \
+}                                                                       \
+}
+
+int main(int argc, char** argv) {
```

Comment:
Should we add a note saying that the C API demo will only work if building the library with consensus turned off? Maybe in this file as well as `demo/c-api/c-api-demo.cc`

Reply:
Which other file are you referring to?

Reply:
Oh oops I referred to the same file 😅 I meant `demo/c-api/README.md`

Reply:
Ah lol ok